### DISCUSSION TITLE
Support Sentry Query Performance Measurement

### DISCUSSION THREAD

**Discussion by thomas-hensel** [prest/prest#873] [API Server] [answered]
Sentry.io or self-hosted Sentry offer query performance measurement. 

It would be great if pREST would support metrics by integrating the Sentry sdk and offer env based configuration for docker deployment. There is a prest/sentry-prestd-plugin repo but it is empty.

Are there any plans to support this in the future?

**Comment by leonardorifeli** [comment]
@thomas-hensel can't you add it directly to your database? In my case, I'dd Sentry self-hosted, getting metrics directly from my PostgreSQL database.

**Comment by avelino** [comment, marked as answer]
I agree with @leonardorifeli that these metrics can be extracted directly from Postgres via the `pg_stat_statements` table (after enabling it in the settings).

However, I believe it is important to support Sentry (not for consuming slow queries), to save the backtrace of prestd when an error occurs - it should be an optional feature, not everyone/team is willing to use Sentry.
-  if `SENTRY_DSN` has content, call Sentry
